airbyte-cdk/python/unit_tests/sources/streams/http/test_http.py

```diff
@@ -253,10 +263,11 @@ def test_raise_on_http_errors_off_non_retryable_4xx(mocker, status_code):
     ),
 )
 def test_raise_on_http_errors(mocker, error):
+    mocker.patch("time.sleep", lambda x: None)
     stream = AutoFailFalseHttpStream()
     send_mock = mocker.patch.object(requests.Session, "send", side_effect=error())
 
-    with pytest.raises(error):
+    with pytest.raises(DefaultBackoffException):
```

Comment:
I would have expected a AirbyteTracedException of type transient_error. Why are we exposing `DefaultBackoffException` here?

Reply:
hmm, this is interesting question...
before migration to httpclient, `RequestException` was raised directly in read_records.
After migration, we try to look for an error in error_mapping: https://github.com/airbytehq/airbyte/blob/18e82d949aa8a9b220f038d903c9872a680bd46b/airbyte-cdk/python/airbyte_cdk/sources/streams/http/error_handlers/http_status_error_handler.py#L50-L53

But there are no errors of this type in `DEFAULT_ERROR_MAPPING`: https://github.com/airbytehq/airbyte/blob/18e82d949aa8a9b220f038d903c9872a680bd46b/airbyte-cdk/python/airbyte_cdk/sources/streams/http/error_handlers/default_error_mapping.py#L11-L26

That is why we face `DefaultBackoffException` here.

Reply:
Oh! I thought this was already done i.e. considering connection issues as transient_errors and retry on these. I guess @pnilan can update us on this once he gets back. In the meanwhile, I have a brain fart: is it surprising that we get a `DefaultBackoffException`? I would have expected all errors to be thrown to be of type `AirbyteTracedException`

Reply:
Agreed, but i can see that this was added intentionally https://github.com/airbytehq/airbyte/blob/18e82d949aa8a9b220f038d903c9872a680bd46b/airbyte-cdk/python/airbyte_cdk/sources/streams/http/error_handlers/http_status_error_handler.py#L53-L61

Generally speaking any (or at least most of them) `Exception` in `requests` library inherit from `RequestException` so it might be easier to check for this error specifically and raise `AirbyteTracedException` in any case.
@pnilan, please provide some context on this check.